### DISCUSSION TITLE
about understanding the parameter "finite_elements" in PFR reactor, kinetic model, adiabatic reactor

### DISCUSSION THREAD

**Discussion by fzerbW** [IDAES/idaes-pse#1152] [Q&A]
Hello all,

I am struggling with the parameter "finite_elements" in PFR reactor model since a few days and hope you can help me with my problems: 

We have impemented a kinetic model and chosen the PFR reactor. For the flowsheet with one single reactor:

`m.fs.R101 = PFR(property_package = m.fs.thermo_params_vapor,
                    reaction_package = m.fs.reaction_properties,
                    has_heat_of_reaction =False,
                    has_equilibrium_reactions = False,
                    has_heat_transfer = False,
                    finite_elements = finite_elements,
                    has_holdup = False,
                    has_pressure_change = False)
`

If we set the parameter **has_heat_transfer = True** and use the default **finite_elements = 20**, the model works very well.  

But when we set **has_heat_transfer = False**, with which we consider it as an adiabatic reactor, we always get the InitializationError of the reactor. Up to now, we still use the default setting finite_elements = 20. 

However, when we set  finite_elements = 1, everything works again. 

So, for my case, I don't need to know what is happening inside the 1D-reactor, I just want the output of the reactor (temperature, pressure, mol fraction, ....) correct. **So my question is:** Do I understand the parameter "finite_elements" correctly? The number of finite_elements only affects the values within the length_domain, not the output. 

I greatly appreciate any help!

**Comment by andrewlee94** [comment]
@fzerbW From your question I am not sure what your understanding of the finite element is, but the number of finite elements has a significant effect on the output final value (unless you are assuming equilibrium reactions in which case a 1-D model is not necessary at all). As conditions very continuously across the reactor length, the number of finite elements determine the accuracy of the model and how closely the discrete numerical solution approximates the real, continuous behaviour.

The first question you should be asking is why the number of finite elements is affecting your degrees of freedom - this implies that you are not fixing the correct conditions in your model and that something is dependent on the numerical approximation (which should never be the case). If you are changing the `has_heat_transfer` setting, you should also need to change the conditions you are fixing as there is now no longer a `heat` term in the energy balances (thus you should have 20 fewer variables). At a guess, I suspect that you might be fixing temperatures or specific enthalpy across the length domain to enforce adiabatic behavior, which you will not need to do as the adiabatic energy balance will enforce the constant specific enthalpy.
